### DISCUSSION TITLE
Clock constraint seems not necessary in Scenario 'PPP'

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#181] [Q&A] [answered]
Hi, 
scenario ppp provided in TUG ftp sets clock parameters as follow:
![image](https://user-images.githubusercontent.com/59271400/233075395-414860b7-41d2-451f-bd1f-0b3532c6ea28.png)
This constraint seems not necessary for transmitter clocks because they are known value provided by sp3 file, which will further be reduced from observation equation and do not cause rank-deficiency. Thus I'd like to confirm wether it's more feasible to set this constrain to '0'? 
Hope someone to help and thanks in advance!
Yongzhao Fan

**Comment by FFFFFG-IGG** [comment]
Note that I've tried PPP in constrain and unconstrain mode, no rank-deficiency occurs, but the position results differ in microm level.
![image](https://user-images.githubusercontent.com/59271400/233228913-ba820b18-4ed5-4a6d-bca3-192670c844b2.png)
This proves that constraint did affect the results, but seems not necessary.

**Comment by tmayerguerr** [comment, marked as answer]
Hi,

The constraints are only applied if all conditions are met

- The respective parameters must estimated
- The respective transmitters and/or receivers must be selected
- The sigma must be > 0

As in your case only transmitters are selected without transmitter clock error estimation , the constraint is not applied here.

regards
Torsten
